code/_onclick/observer_onclick.dm

```diff
@@ -28,6 +28,9 @@
 		return
 	if(world.time <= next_move)
 		return
+	if(gas_analyzer && istype(A, /obj/machinery/atmospherics/pipe))
```

Comment:
Honestly I'd just say to move this onto `attack_ghost()` for pipes

Reply:
I'm assuming you mean set it up as 
```/obj/machinery/atmospherics/pipe/attack_ghost(mob/user)
	gas_analyzer()
``` 
With it setup like this though, I can't call the new proc I setup on /mob/dead/observer, and I'm not really sure where under pipes I should or would setup a new proc for anaylzing. Where I currently placed it does seem out of place though.